cuqi/distribution/_implicit_regularized_gaussian.py

```diff
@@ -0,0 +1,273 @@
+from cuqi.utilities import get_non_default_args
+from cuqi.distribution import Distribution, Gaussian, GMRF
+from cuqi.solver import ProjectNonnegative, ProjectBox, ProximalL1
+
+import numpy as np
+
+
+class ImplicitRegularizedGaussian(Distribution):
+    """ Implicit Regularized Gaussian distribution.
+
+    Defines a Gaussian distribution with implicit regularization. The regularization can be defined
+    in the form of a proximal operator or a projector. Alternatively, preset constraints and regularization
+    can be used.
+
+    Only one of proximal, projector, constraint or regularization can be provided. If none of them are provided,

```

Comment:
We always have a default set of parameters other than scaler and vector. Considering it is a regularized gaussian, it would be a bit strange to have the default be no regularization? Then I would rather actually force the user to specify a type of regularization. Thoughtts @amal-ghamdi ?

Reply:
It could be better to require the user to specify regularization so they are aware of the impact. But this is just my suggestion, you can alternatively leave it as it is if you think that is better.

Reply:
I will let @jakobsj  weight in. This would be the first distribution in CUQIpy that does not a default argument for non-vector type parameters that can also be None. E.g. we can do

```python
x = Gaussian(0, 1)
y = GMRF(0, 1)
```
etc.

Reply:
Just saw this, please make sure this is resolved ie no default is used. Perhaps just the docstring needs updating in line with the gaussian version.

Reply:
Just docstring yes